src/Nimiq/EdPrivateKey.h

```diff
@@ -0,0 +1,63 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include <array>
+#include <vector>
+
+namespace TW {
+namespace Nimiq {
+
+class EdPrivateKey {
```

Comment:
Multiple Blockchains need something like this, let's not introduce a new PrivateKey.

Reply:
@alejandro-isaza I was actually considering opening an issue just for this.
The existing `PrivateKey.h` only handles ECDSA signatures.

Even though the private key sizes are the same for _ECDSA secp256k1_ and _Ed25519_, it seemed wrong to mix logic in the same file, which is why I created this new PrivateKey class.

How about creating a file like `src/PrivateKeyEdwards.h` instead?

Reply:
That sounds good, but would call it just `PrivateKey` and put it under a `ed25519` namespace while moving the current one to `secp256k1` namespace. That way you can do `using ed25519` and just use PrivateKey from then on.

Reply:
@alejandro-isaza I've been working on this but the PR is getting huge.

I'm adding `Secp256k1` to every exported symbol in the C API right now. Is there a better way?

Reply:
I can do this, we need it for other blockchains as well. Are you ok to wait for me to do it? It will probably be tomorrow.

Reply:
@alejandro-isaza Oops, I read that to late. My PR is almost working, but it's a bit large. I can totally understand if you want to redo it on your own instead. Also, can we talk on Telegram about it?

Reply:
See #87

Reply:
Fixed in #94